Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -88,8 +229,41 @@ public static int[] InsertLast(int[] array, int number)
         /// <returns>A new array with element inserted at a given index. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertAt(int[] array, int number, int index)
         {
-            // ToDo: implement.
-            return array;
+            if (index < 0 || index > array?.Length)
+            {
+                return array;
+            }
+            if (ArrayIsNullOrEmpty(array))
+            {
+                return new int[] { number };
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4930349)